storage/adapter.go

```diff
@@ -0,0 +1,407 @@
+package storage
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-sectorbuilder"
+	s2 "github.com/filecoin-project/go-storage-miner"
+	"github.com/ipfs/go-cid"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/lotus/build"
+	"github.com/filecoin-project/lotus/chain/actors"
+	"github.com/filecoin-project/lotus/chain/events"
+	"github.com/filecoin-project/lotus/chain/types"
+)
+
+type TicketFn func(context.Context) (*sectorbuilder.SealTicket, error)
+
+type StorageMinerNodeAdapter struct {
+	api    storageMinerApi
+	events *events.Events
+
+	maddr address.Address
+	waddr address.Address
+
+	tktFn TicketFn
+}
+
+func NewStorageMinerNodeAdapter(n storageMinerApi, evts *events.Events, maddr, waddr address.Address, tktFn TicketFn) *StorageMinerNodeAdapter {
+	return &StorageMinerNodeAdapter{api: n, events: evts, tktFn: tktFn, maddr: maddr, waddr: waddr}
+}
+
+func (m *StorageMinerNodeAdapter) SendSelfDeals(ctx context.Context, pieces ...s2.PieceInfo) (cid.Cid, error) {
```

Comment:
With the specs-actors package we could merge all of those Send methods into one - `Send(from, to Address, method abi.MethodNum, value big.Int, params []byte)`.

Reply:
Sounds good. Note that delivering a storage miner which uses such a method is blocked on some work which @hannahhoward is doing which will allow us to map from CommP (a 32-byte array) to a CID for purposes of constructing a [`DealProposal`](https://github.com/filecoin-project/specs-actors/blob/48d5262d247ff285e74ce3ae48819eb5e92f86a5/actors/builtin/market/deal.go#L21).